wordpress-mobile/WordPress-iOS
# Feature/7675 password challenge

Refs #7675
This PR wires up the password challenge step shown when a user attempts to login via google but a matching wpcom account has not yet been connected.

Prep: 
- Create a Google account. 
- Create a wpcom account with the google account's email address.

To test:
- Via a browser, log into wpcom and view your account settings page. Choose the Social Login tab and confirm the account is not connected.
- On the email login screen, tap to log in via Google. 
- Authenticate with the Google account you created.
- Confirm the password screen in shown and shows the text from the mock up.
- Authenticate to wpcom.
- Via a browser, log into wpcom and view your account settings page. Choose the Social Login tab and confirm the account *is* connected.

Confirm that all tests pass.

Needs review: @nheagy
